### PR TITLE
Run qrexec daemon and redirect rpc calls to stubdom

### PR DESCRIPTION
**Pull request by easydozen**
QubesOS/qubes-issues#5802

### CODE REVIEW

File: qubes/vm/qubesvm.py [hunk 1]
```diff
@@ -1629,14 +1634,19 @@ def start_daemon(*command, input=None, **kwargs):
...
         """Start qrexec daemon.
 
         :raises OSError: when starting fails.
         """
 
         self.log.debug('Starting the qrexec daemon')
-        qrexec_args = [str(self.xid), self.name, self.default_user]
+        if stubdom:
+            qrexec_args = [str(self.stubdom_xid), self.name + '-dm', \
+                            self.default_user]
```
Comment (marmarek) [inline qubes/vm/qubesvm.py:1646]: You sure? I don't think there are any non-root users in stubdomain...

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarek** [comment]
And also please add tests for the newly introduced cases.

**Comment by easydozen** [comment]
> And also please add tests for the newly introduced cases.
I've added one subtest for run_service here. Another one provided by QubesOS/qubes-app-linux-usb-proxy/pull/22.
Did I miss something?

File: qubes/vm/qubesvm.py [hunk 2]
```diff
@@ -1445,28 +1448,30 @@ async def run_service(self, service, source=None, user=None,
...
             # XXX what about autostart?
             raise qubes.exc.QubesVMNotRunningError(
-                self, 'Domain {!r} is paused'.format(self.name))
+                self, 'Domain {!r} is paused'.format(name))
         if not self.is_running():
             if not autostart:
                 raise qubes.exc.QubesVMNotRunningError(self)
             await self.start(start_guid=gui)
 
         if not self.is_qrexec_running():
```
Comment (marmarek) [inline qubes/vm/qubesvm.py:1465]: This is a bit unfortunate - it checks if qrexec is connected to the actual domain, not to the stubdomain, even if the call is requested to the stubdomain (`stubdom=True`).

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarek** [comment]
And now this needs `Requires: xen-hvm-stubdom-linux-full` in the spec file.

File: rpm_spec/core-dom0.spec.in [hunk 3]
```diff
@@ -82,7 +82,7 @@ Conflicts:      qubes-gui-dom0 < 1.1.13
 %if "x%{?backend_vmm}" == "xxen"
 Requires:       xen-runtime
 Requires:       xen-hvm
-Requires:       xen-hvm-stubdom-linux >= 1.0.13
```
Comment (marmarek) [inline rpm_spec/core-dom0.spec.in:85]: No, both are needed

**Review by marmarek — APPROVED** [review]

**Pull request merged**